calendar_resource/tests/setup.py

```diff
@@ -0,0 +1,117 @@
+
+        self.event_type_4 = self.env.ref('calendar.categ_meet4')
+        self.event_type_5 = self.env.ref('calendar.categ_meet5')
+
+        self.leave_1 = self.env.ref(
+            'resource.resource_analyst_leaves_demoleave1'
+        )
+
+        self.env.user.partner_id.tz = 'UTC'
+        self.env.user.lang = 'en_US'
+
+        self.Calendar = self.env['resource.calendar']
+        self.Event = self.env['calendar.event']
+
+        self.intervals = [
+            ('2017-03-07 00:00:00', '2017-03-07 16:00:00'),
+            ('2017-03-07 12:00:00', '2017-03-07 20:00:00'),
+            ('2017-03-07 20:00:00', '2017-03-07 23:59:59'),
+            ('2017-03-08 00:00:00', '2017-03-08 16:00:00'),
+            ('2017-03-08 05:00:00', '2017-03-08 11:30:00'),
+            ('2017-03-09 09:00:00', '2017-03-09 23:59:00'),
+        ]
+        self.intervals = self._intervals_to_dt(self.intervals)
+
+    def _intervals_to_dt(self, intervals):
+        """ Converts all intervals from string values to datetime.
+
+        Args:
+            intervals (list): List of tuples each containing
+                a start and stop string value to be converted
+                to datetime.
+
+        """
+        for index, interval in enumerate(intervals):
+            intervals[index] = (
+                fields.Datetime.from_string(interval[0]),
+                fields.Datetime.from_string(interval[1]),
+            )
+        return intervals
+
+    def _get_datetime_interval(self, start_weekday, start_time,
+                               end_weekday, end_time):
+        """ Use this method to ensure events are always in the future """
+        start_date = fields.Date.from_string(
+            fields.Datetime.now()
+        )
+        # 35 = 5 weeks * 7 days per week
+        # done to avoid demo resource leaves
+        # that may be within 7 days of event
+        start_date += timedelta(days=35)
+        time_format = '%H:%M:%S'
+        start_time = datetime.strptime(start_time, time_format).time()
+        end_time = datetime.strptime(end_time, time_format).time()
+
+        if not start_date.weekday() == 0:
+            while start_date.weekday() != 0:
+                start_date -= timedelta(days=1)
+
+        while start_date.weekday() != start_weekday:
+            start_date += timedelta(days=1)
+
+        end_date = start_date
+        while end_date.weekday() != end_weekday:
+            end_date += timedelta(days=1)
+
+        start_datetime = fields.Datetime.to_string(
+            datetime.combine(start_date, start_time)
+        )
+        end_datetime = fields.Datetime.to_string(
+            datetime.combine(end_date, end_time)
+        )
+
+        return (start_datetime, end_datetime)
+
+    def _create_event(self, vals=None):
+        create_vals = {
+            'name': 'Test Event',
+            'resource_ids': [(6, 0, [self.resource_1.id])],
+            'categ_ids': [(6, 0, [self.event_type_4.id])],
+            'allday': True,
```

Comment:
Changed the tests to specifically define allday in each test for better clarity.